io.openems.backend.uiwebsocket/src/io/openems/backend/uiwebsocket/impl/OnRequest.java

```diff
@@ -435,7 +435,7 @@ private CompletableFuture<? extends JsonrpcResponseSuccess> handleGetUserAlertin
 		var edgeId = request.getEdgeId();
 		List<AlertingSetting> users;
 
-		if (user.getRole(edgeId).orElse(Role.GUEST).isLessThan(Role.ADMIN)) {
+		if (user.getRole(edgeId).orElse(user.getGlobalRole()).isLessThan(Role.ADMIN)) {
```

Comment:
This and the change in L460 means, that a user, who does not have a role for an edge, can use his global rights for the edge. I think of a usecase where an installer, which has no access to an edge, can access this edge with this installer role. 
As we are embedded in the OpenEMS UI security concept, this is not a problem. 

But maybe there are ways to use the Backend JSON-RPC interface to use this weakness.

@da-Kai Is it really needed to increase the rights here? If so, do you see any security weakness here?

Reply:
@clehne it rather bothers me that I used a low-fixed-value as fallback (_"Defaulting"_) instead of Optional::isPresent.

i would prefer to check whether the user has admin rights using the following method:
`var userIsAdmin = user.getRole(edgeId).map(role -> role.isAtLeast(Role.ADMIN)).orElse(false);`
and to rewrite the if-statements with it.

Reply:
Changed this for get and set UserAlertingSettings

Thank you!